src/core/server/models/user/user.ts

```diff
@@ -671,6 +682,63 @@ export async function updateUserPassword(
   return result.value || null;
 }
 
+export async function scheduleDeletionDate(
+  mongo: Db,
+  tenantID: string,
+  userID: string,
+  deletionDate: Date
+) {
+  const users = collection(mongo);
+  const result = await users.findOneAndUpdate(
+    {
+      id: userID,
+      tenantID,
+    },
+    {
+      $set: {
+        scheduledDeletionDate: deletionDate,
+      },
+    },
+    {
+      returnOriginal: false,
```

Comment:
Ensure you retains the comment about the `returnOriginal` variable, as it's pretty unclear.